server/rest/src/main/java/org/infinispan/rest/resources/StaticContentResource.java

```diff
@@ -80,9 +80,7 @@ private File resolve(String resource) {
       if (uri.equals(noFileUri)) {
          return completedFuture(responseBuilder.location(rootUri).status(MOVED_PERMANENTLY).build());
       }
-
-      String resource = uri.equals(rootUri) ? DEFAULT_RESOURCE : uri.substring(uri.indexOf(urlPath) + urlPath.length() + 1);
-
+      String resource = uri.equals(rootUri) || (uri.contains(rootUri) && !uri.contains("."))? DEFAULT_RESOURCE : uri.substring(uri.indexOf(urlPath) + urlPath.length() + 1);
```

Comment:
Not sure how to parse this, maybe you can extract a variable so the name is a hint, and add a test with a "." in the uri as an example?

Reply:
there is already a test that covers .

Reply:
ok for extracting to a variable